access/handler.go

```diff
@@ -1115,13 +1115,11 @@ func (h *Handler) SendAndSubscribeTransactionStatuses(
 
 	sub := h.api.SubscribeTransactionStatuses(ctx, &tx, request.GetEventEncodingVersion())
 
-	messageIndex := counters.NewMonotonousCounter(0)
+	messageIndex := counters.NewMonotonousCounter(1)
 	return subscription.HandleSubscription(sub, func(txResults []*TransactionResult) error {
 		for i := range txResults {
 			value := messageIndex.Value()
-			if ok := messageIndex.Set(value + 1); !ok {
-				return status.Errorf(codes.Internal, "the message index has already been incremented to %d", messageIndex.Value())
-			}
+			messageIndex.Increment()
```

Comment:
there's a race condition where between 
```golang
value := messageIndex.Value()
```
and
```golang
messageIndex.Increment()
```

consider there are 2 responses received at the same time:
1. Response 1 calls `messageIndex.Value()`
2. Response 2 calls `messageIndex.Value()` immediately afterwards
3. Both now have the same value for `value`
4. Next both will increment the counter, skipping an index.

Since `messageIndex` is intended for clients to detect that they missed a message, it's important for it to be sequential.

Instead, you can just use 
```golang
messageIndex := counters.NewMonotonousCounter(0)
	...
	value := messageIndex.Increment()
```

Since increment returns the "new" value, it will start from 1, but is guaranteed to have sequential values

Reply:
My bad. Fixed